builtin/providers/azurerm/resource_arm_servicebus_queue.go

```diff
@@ -0,0 +1,229 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+
+	"github.com/Azure/azure-sdk-for-go/arm/servicebus"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceArmServiceBusQueue() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmServiceBusQueueCreate,
+		Read:   resourceArmServiceBusQueueRead,
+		Update: resourceArmServiceBusQueueCreate,
+		Delete: resourceArmServiceBusQueueDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"namespace_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"location": locationSchema(),
+
+			"resource_group_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"auto_delete_on_idle": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+
+			"default_message_ttl": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+
+			"duplicate_detection_history_time_window": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"enable_batched_operations": {
+				Type:     schema.TypeBool,
+				Optional: true,
```

Comment:
We should provide a default here. We have a known bug in Terraform that if you specify a value of `true` then change back to `false` then terraform won't get that change

this is because of d.GetOk("false") believing that the value doesn't exist as the default value for a bool is false